src/dsp_tools/commands/excel2json/resources.py

```diff
@@ -112,63 +112,77 @@ def _row2resource(
-        try:
-            [int(float(x)) for x in details_df["gui_order"]]
-        except ValueError:
-            validation_passed = False
-    else:  # column gui_order present but not properly filled in (missing values)
-        validation_passed = False
-    if not validation_passed:
-        raise UserError(
-            f"Sheet '{name}' in file 'resources.xlsx' has invalid content in column 'gui_order': "
-            f"only positive integers allowed (or leave column empty altogether)"
+    if len(class_property_df) == 0:
+        warnings.warn(
+            f"Sheet '{class_name}' in file 'resources.xlsx' does not have any properties listed.\n"
+            f"Creation of the resource class continues without 'cardinalities' section."
         )
+        return []
+    cards = _create_all_property_restrictions(class_name, class_property_df)
+    return cards
+
 
+def _create_all_property_restrictions(class_name: str, class_property_df: pd.DataFrame) -> list[dict[str, str | int]]:
+    class_property_df = _check_complete_gui_order(class_name, class_property_df)
     cards = []
-    for i, detail_row in details_df.iterrows():
-        index = int(str(i))  # j is a label/index/hashable, but we need an int
-        gui_order = detail_row.get("gui_order", "")
-        gui_order = regex.sub(r"\.0+", "", str(gui_order))
+    for i, detail_row in class_property_df.iterrows():
         property_ = {
             "propname": ":" + detail_row["property"],
             "cardinality": detail_row["cardinality"].lower(),
-            "gui_order": int(gui_order or index + 1),  # if gui_order not given: take sheet order
+            "gui_order": detail_row["gui_order"],
         }
         cards.append(property_)
+    return cards
 
-    # build the dict structure of this resource and append it to the list of resources
-    resource = {"name": name, "super": supers, "labels": labels}
-    if comments:
-        resource["comments"] = comments
-    resource["cardinalities"] = cards
 
-    return resource
+def _check_complete_gui_order(class_name: str, class_property_df: pd.DataFrame) -> pd.DataFrame:
+    detail_problem_msg = ""
+    if "gui_order" not in class_property_df:
+        detail_problem_msg = "The column 'gui_order' does not exist."
+    elif class_property_df["gui_order"].isna().any():
+        detail_problem_msg = "Some rows are empty.\n"
+
+    if not detail_problem_msg:
+        try:
+            class_property_df["gui_order"] = [int(float(x)) for x in class_property_df["gui_order"]]
+            return class_property_df
+        except ValueError:
+            detail_problem_msg = "Some rows contain invalid characters that could not be converted to an integer."
+
+    class_property_df["gui_order"] = [x for x in range(1, len(class_property_df) + 1)]
+
+    complete_msg = (
+        f"Sheet '{class_name}' in file 'resources.xlsx' has invalid content in column 'gui_order'\n"
+        f"{detail_problem_msg}\n"
```

Comment:
If the column gui_order does not exist, the user will receive a confusing message:

"The sheet xyz in file xyz has invalid content in column gui_order. The column 'gui_order' does not exist."